nltools/stats.py

```diff
@@ -1008,6 +1019,40 @@ def regress(X, Y, mode='ols', **kwargs):
     return b.squeeze(), t.squeeze(), p.squeeze(), df.squeeze(), res.squeeze()
 
 
+def regress_permutation(X, Y, n_permute=5000, random_state=None, **kwargs):
+    """
+    Permuted regression. Permute the design matrix each time by shuffling rows before running the estimation.
+
+    Args:
+        X (ndarray): design matrix; assumes intercept is included
+        Y (ndarray): dependent variable array; if 2d, a model is fit to each column of Y separately
+        n_permute: (int) number of permutations
+        tail: (int) either 1 for one-tail or 2 for two-tailed test (default: 2)
+        n_jobs: (int) The number of CPUs to use to do the computation. -1 means all CPUs.
+        kwargs: optional argument to regress()
+
+    """
+
+    random_state = check_random_state(random_state)
+    b, t = regress(X, Y, stats='tstats', **kwargs)   
+    p = np.zeros_like(t)
+
+    if (X.shape[1] == 1) and (all(X[:] == 1.)):
+        func = lambda x: (x.squeeze() * random_state.choice([1, -1], x.shape[0]))[:, np.newaxis]
+    else:
+        func = random_state.permutation
+    
+    # We could optionally Save (X.T * X)^-1 * X.T so we dont have to invert each permutation, but this would require not relying on regress() and because the second-level design mat is probably on the small side we might not actually save that much time
+    # inv = np.linalg.pinv(X)
+
+    for i in range(n_permute):
+        _, _t = regress(func(X.values), Y, stats='tstats', **kwargs)
+        p += np.abs(_t) >= np.abs(t)
```

Comment:
Does this mean this method can only do one-sided permutation tests?

Reply:
Just changed this so we can do both 1 and 2 tailed

Reply:
Also added thresholding